storage_controller/src/reconciler.rs

```diff
@@ -846,6 +854,40 @@ impl Reconciler {
         }
     }
 
+    /// Compare the observed state snapshot from when the reconcile was created
+    /// with the final observed state in order to generate observed state deltas.
+    pub(crate) fn observed_deltas(&self) -> Vec<ObservedStateDelta> {
+        let mut deltas = Vec::default();
+
+        for (node_id, location) in &self.observed.locations {
+            let previous_location = self.original_observed.locations.get(node_id);
+            match previous_location {
+                Some(prev) => {
+                    if location.conf != prev.conf {
+                        deltas.push(ObservedStateDelta::Upsert(Box::new((
```

Comment:
maybe it might make more sense to return a `bool_var`, and then put the `deltas.push` code into an `iff bool_var`?

Reply:
Sure. Done in https://github.com/neondatabase/neon/pull/9276/commits/9a415528539dabe386a4e1c15b3963536a469ca8